tests/test_context.py

```diff
@@ -48,3 +52,49 @@ def test_message_handler(device):
     assert context.message_handler is message_handler_test
     with context:
         assert current_context().message_handler is context.message_handler
+
+
+def test_async_wake_interval(device):
+    """Test read-write property async_wake_interval."""
+    with Context(device) as context:
+        context.async_wake_interval = 42
+        assert context.async_wake_interval == 42
+
+
+def test_default_resampler_index(device):
+    """Test return values default_resampler_index"""
+    with Context(device) as context:
+        assert context.default_resampler_index >= 0
+        index = context.default_resampler_index
+        assert len(context.available_resamplers) > index
+
+
+def test_doppler_factor(device):
+    """Test write property doppler_factor."""
+    with Context(device) as context:
+        context.doppler_factor = 4/9
+        context.doppler_factor = 9/4
+        context.doppler_factor = 0
+        context.doppler_factor = inf
+        with raises(ValueError): context.doppler_factor = -1
+
+
+def test_speed_of_sound(device):
+    """Test write property speed_of_sound."""
+    with Context(device) as context:
+        context.speed_of_sound = 5/7
+        context.speed_of_sound = 7/5
+        with raises(ValueError): context.speed_of_sound = 0
+        context.speed_of_sound = inf
+        with raises(ValueError): context.speed_of_sound = -1
+
+
+def test_distance_model(device):
+    """Test preset values distance_model"""
+    distance_model = 'inverse clamped'
+    distance_model = 'linear clamped'
+    distance_model = 'exponent clamped'
+    distance_model = 'inverse'
+    distance_model = 'linear'
+    distance_model = 'exponent'
+    distance_model = 'none'
```

Comment:
Shouldof been
```python
    with Context(device) as context:
        for model in distance_models: context.distance_model = model
        with raises(ValueError): context.distance_model = 'some random string that comes up to your mind'
```

Reply:
wait, sth wrong...
pytest responded:
```
device = Device('Built-in Audio Analog Stereo')

    def test_distance_model(device):
        """Test preset values distance_model."""
        with Context(device) as context:
>           for model in distance_models: context.distance_model = model
E           NameError: name 'distance_models' is not defined
```

Reply:
import it from palace then

Reply:
I leave the link here, check it:
https://youtu.be/NuyEcB8Vcq8

Reply:
Have you reinstalled palace (from this branch)?

Reply:
Okay now I have already reinstalled and I got this:
```
device = Device('Built-in Audio Analog Stereo')

    def test_distance_model(device):
        """Test preset values distance_model."""
        with Context(device) as context:
>           for model in distance_models: context.distance_models = model
E           AttributeError: 'palace.Context' object has no attribute 'distance_models'

tests/test_context.py:95: AttributeError
```
Also, I have already install palace by `CYTHON_TRACE=1 ./setup.py install --user` but it still gives me this:
```
/home/meteora/.local/lib/python3.6/site-packages/pytest_cov/plugin.py:254: PytestWarning: Failed to generate report: No data to report.

  self.cov_controller.finish()
```

Reply:
oof, I resolved the upper one, it just me adding "s" in wrong place...